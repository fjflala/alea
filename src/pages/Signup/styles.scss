.signup {
  padding: 32px;
  height: 100%;
  background-color: rgb(245,85,46);

  &__card {
    max-width: 480px;
    margin: 32px auto;
  }

  &__alea-logo {
    display: block;
    max-width: 80px;
    text-align: center;
    margin: 16px auto;
  }

  &__form {
    text-align: center;
  }

  &__button {
    margin: 16px auto;
  }

  &__error-message {
    color: red;
  }
}