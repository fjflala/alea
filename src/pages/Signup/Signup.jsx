/**
 * Modules dependencies
 */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

/**
 * Actions
 */
import { authRegister } from '../../modules/Authentication/actions';

/**
 * UI Component
 */
import Card from '../../components/Card';
import Input from '../../components/Input';
import Button from '../../components/Button';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents a view of signup
 */
export const Signup = ({
  history,
  isAuthenticated,
  token,
  error,
  register,
}) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState(null);

  const onSubmit = (event) => {
    event.preventDefault();

    register({
      data: {
        email,
        password,
      },
    });
  };

  useEffect(() => {
    if (isAuthenticated && token.length > 0) {
      history.push('/users');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isAuthenticated, token]);

  return (
    <section className="signup">
      <Card className="signup__card">
        <img
          src="https://www.alea.com/wp-content/themes/alea/library/images/logo-alea.svg"
          alt="Alea"
          className="signup__alea-logo"
        />
        {error.length > 0 && (
          <p className="signup__error-message">
            The register procces is mocked.
            <br />
            You can test with the mocked values
          </p>
        )}
        <form onSubmit={onSubmit}>
          <Input type="text" placeholder="email" onChange={e => setEmail(e.target.value)} />
          <Input type="password" placeholder="password" onChange={e => setPassword(e.target.value)} />
          <Button type="submit" modifier="primary" className="signup__button">Signup</Button>
        </form>
        <p>
          If you already have an account, please &nbsp;
          <Link to="/login">login</Link>
        </p>
      </Card>
      <Card className="signup__card">
        <h4>Mocked Values</h4>
        <strong>User:</strong>
        &nbsp;eve.holt@reqres.in
        <br />
        <strong>Password:</strong>
        &nbsp;pistol
      </Card>
    </section>
  );
};

/**
 * PropTypes definitions 
 */
Signup.propTypes = {
  history: PropTypes.shape({}).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  token: PropTypes.string,
  error: PropTypes.string,
  register: PropTypes.func.isRequired,
};

Signup.defaultProps = {
  error: '',
  token: '',
};
 
/**
 * Connect to store
 */
const mapStateToProps = state => ({
  isAuthenticated: state.authentication.isAuthenticated,
  token: state.authentication.token,
  error: state.authentication.error,
});

const mapDispatchToProps = {
  register: authRegister,
};

/**
 * Expose component
 */
export default connect(mapStateToProps, mapDispatchToProps)(Signup);
