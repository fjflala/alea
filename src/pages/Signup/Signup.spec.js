import React from 'react';
import renderer from 'react-test-renderer';
import { Signup } from './Signup';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { mount } from 'enzyme';

describe('<Signup />', () => {
  let props;
  
  beforeEach(() => {
    props = {
      isAuthenticated: false,
      token: '',
      error: '',
      register: jest.fn(),
      history: {},
    };
  });

  it('should correctly render', () => {
    const tree = renderer
      .create(<Router history={createMemoryHistory({route: ['/signup']})}>
        <Signup {...props} />
      </Router>)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('should call register method when click on register button', () => {
    const wrapper = mount(<Router history={createMemoryHistory({route: ['/signup']})}>
      <Signup {...props} />
    </Router>);

    wrapper.find('form').simulate('submit');
    
    expect(props.register).toHaveBeenCalled();
  });

  it('should show an error message', () => {
    props.error = 'Error while trying to register user';
    const wrapper = mount(<Router history={createMemoryHistory({route: ['/signup']})}>
      <Signup {...props} />
    </Router>);
    
    expect(wrapper.find('.signup__error-message').length).toBe(1);
  });
});