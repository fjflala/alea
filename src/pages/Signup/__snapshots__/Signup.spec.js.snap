// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should correctly render 1`] = `
<section
  className="signup"
>
  <div
    className="ui-card signup__card"
  >
    <img
      alt="Alea"
      className="signup__alea-logo"
      src="https://www.alea.com/wp-content/themes/alea/library/images/logo-alea.svg"
    />
    <form
      onSubmit={[Function]}
    >
      <input
        className="ui-input "
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="email"
        type="text"
      />
      <input
        className="ui-input "
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="password"
        type="password"
      />
      <button
        className="ui-button ui-button--primary signup__button"
        onClick={[Function]}
        type="submit"
      >
        Signup
      </button>
    </form>
    <p>
      If you already have an account, please  
      <a
        href="/login"
        onClick={[Function]}
      >
        login
      </a>
    </p>
  </div>
  <div
    className="ui-card signup__card"
  >
    <h4>
      Mocked Values
    </h4>
    <strong>
      User:
    </strong>
     eve.holt@reqres.in
    <br />
    <strong>
      Password:
    </strong>
     pistol
  </div>
</section>
`;
